pkg/replica/server.go

```diff
@@ -113,7 +115,21 @@ func (s *Server) Status() (State, Info) {
 		info, err := ReadInfo(s.dir)
 		if os.IsNotExist(err) {
 			return Initial, Info{}
-		} else if err != nil {
+		}
+		if errors.Is(err, io.EOF) {
+			volumeMetaFilePath := (&Replica{dir: s.dir}).diskPath(volumeMetaData)
+
+			fileInfo, statErr := os.Stat(volumeMetaFilePath)
+			if statErr != nil {
+				logrus.Errorf("Failed to read info in replica directory %s: %v", s.dir, err)
+				return Error, Info{}
+			}
+
+			if fileInfo.Size() == 0 {
+				return Initial, Info{}
+			}
```

Comment:
Suggest change to below.

```go
replica := Replica{dir: s.dir}
if er := replica.checkValidVolumeMetaData(); err != nil {
    return Initial, Info{}
}
```

In `checkValidVolumeMetaData`, you can try to decode the file to see if it's valid.

Reply:
Modified. Add `checkValidVolumeMetaData` to check if volume metadata file is valid.